components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/util/ActionManagementUtil.java

```diff
@@ -64,4 +71,66 @@ public static ActionMgtServerException handleServerException(
 
         return new ActionMgtServerException(error.getMessage(), description, error.getCode(), e);
     }
+
+    /**
+     * Validate whether required fields exist.
+     *
+     * @param field Field value.
+     * @throws ActionMgtClientException if the provided field is empty.
+     */
+    public static void isFieldEmpty(String field) throws ActionMgtClientException {
+
+        if (StringUtils.isBlank(field)) {
+            throw ActionManagementUtil.handleClientException(ActionMgtConstants.ErrorMessages.
+                    ERROR_EMPTY_ACTION_REQUEST_FIELD);
+        }
+    }
+
+    /**
+     * Validate the action name.
+     *
+     * @param name Action name.
+     * @throws ActionMgtClientException if the name is not valid.
+     */
+    public static void isValidActionName(String name) throws ActionMgtClientException {
+
+        Pattern regexPattern = Pattern.compile(ACTION_NAME_REGEX);
+        boolean isValidName = regexPattern.matcher(name).matches();
+        if (!isValidName) {
+            throw ActionManagementUtil.handleClientException(ActionMgtConstants.ErrorMessages.
```

Comment:
Each validation should just throw the client error. It's the API layer that handles the errors

Reply:
`handleClientException` method only returns the client exception and does not handle it here at the service layer. It is used to construct the exception properly.

This method was already present in the `ActionManagementUtil` class. IMO the confusion arises due to the method name and we could consider renaming it to better reflect its functionality.

cc: @malithie